tests/contrib/pyramid/test_pyramid.py

```diff
@@ -58,6 +62,36 @@ def test_404(self):
         eq_(s.meta.get('http.status_code'), '404')
         eq_(s.meta.get('http.url'), '/404')
 
+    def test_302(self):
```

Comment:
Thank you very much for these tests! Actually I'm figuring out how we can improve it a bit, since it seems that your new code path is not hit by this test (if I remove the exception handling, the test still pass). Let me do some investigation from my side, just to be sure I'm not missing something.

Thank you!

Reply:
@TylerLubeck can you provide me more details about this issue? because I can't reproduce it entirely with a test. Would be great to iterate over this problem so I can merge it ASAP.

Thank you!

Reply:
Interesting. Are you seeing the same for the 204 version? What do you mean by removing the exception handling?

Solid chance I need to be doing `raise HTTPFound()` instead of `raise HTTPFound`. I can verify this in a few hours

Reply:
Yes actually if I undo the patch (so removing only the `except HTTPException as e:` block) and run:
```
$ tox -e "{py27,py34,py35,py36}-pyramid{17,18}-webtest"
```

I get all tests passing. Can you verify if it happens to you too or if it's something else? Thank you very much!

Reply:
Hello @TylerLubeck ! Let me know if you need help to triage better this issue. Would be great having a regression test so we can merge this PR immediately! Thank you!